### ISSUE TITLE
NPE with ViewStub.

### ISSUE THREAD

**Issue by jmodrako** [ppicas/custom-typeface#2]
First of all I'd like to say big thank you for that library! But, I encountered problem. Here is stack-trace:

`Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.TypedArray android.content.Context.obtainStyledAttributes(android.util.AttributeSet, int[], int, int)' on a null object reference
            at android.view.ViewStub.<init>(ViewStub.java:105)
            at android.view.ViewStub.<init>(ViewStub.java:101)
            at android.view.ViewStub.<init>(ViewStub.java:96)
            at java.lang.reflect.Constructor.newInstance(Native Method)
            at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
            at android.view.LayoutInflater.createView(LayoutInflater.java:607)
            at cat.ppicas.customtypeface.CustomTypeface.createView(CustomTypeface.java:279)
            at modrakowski.pl.cartracker.app.MainActivity.onCreateView(MainActivity.java:114)`

**Comment by jmodrako** [comment]
The problem is with that `ViewStub` is not part of `android.widget` package. Look here: https://developer.android.com/reference/android/widget/package-summary.html

And here in your code: `if (name.indexOf('.') == -1) {prefix = "android.widget.";}`

**Comment by ppicas** [comment]
Hi Jacek, thanks for your comment. Sorry for the delay but I have been a little bit disconnected from my side projects.

I tried to use a `ViewStub` on a layout and I couldn't reproduce the bug. And I'd discard the problem to be that `ViewStub` is not part of `android.widget` because on that case `createView` will return `null` and `LayoutInflater` will try the default approach to create and instance of `ViewStub`.

Taking a look to the stack trace that you pasted, I realized that the NPE is thrown because `Context` passed to `ViewStub` constructor is null. With this we can also discard the problem to be located at `createView` method. Maybe if you still have this problem, you could paste some parts of your code to detect why `LayoutInflater` is passing a null `Context`.

Thanks for your support!

**Comment by jmodrako** [comment]
I just did the workaround for this issue like this one:

`@Override public View onCreateView(String name, Context context, AttributeSet attrs) {
        return ViewStub.class.getSimpleName().contains(name) ? super.onCreateView(name, context, attrs)
                : CustomTypeface.getInstance().createView(name, context, attrs);
    }`

So this code omnit creating the ViewStub through CustomTypeface.getInstance().createView and it's fine.

**Comment by Advice-Dog** [comment]
Hi jmodrako and ppicas,

I've been running into the same issue, but only on devices that are running 5.0.2 (Nexus 7, emulator) after going from the first activity to the second. However, on my phone running 4.1.1, it doesn't seem to have any issues.

I've also tried jmodrako's work around, but that doesn't have any effect on the crash. 

Any ideas on how to fix this issue?
Thanks.

```
03-12 16:05:37.193  26020-26020/myapp.com.app E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: myapp.com.app, PID: 26020
    java.lang.RuntimeException: Unable to start activity ComponentInfo{myapp.com.app/myapp.com.app.Signup}: android.view.InflateException: Binary XML file line #29: Error inflating class <unknown>
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2298)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
            at android.app.ActivityThread.access$800(ActivityThread.java:144)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
     Caused by: android.view.InflateException: Binary XML file line #29: Error inflating class <unknown>
            at android.view.LayoutInflater.createView(LayoutInflater.java:633)
            at cat.ppicas.customtypeface.CustomTypeface.createView(CustomTypeface.java:285)
            at cat.ppicas.customtypeface.CustomTypeface.onCreateView(CustomTypeface.java:307)
            at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:806)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:504)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:414)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:365)
            at com.android.internal.policy.impl.PhoneWindow.generateLayout(PhoneWindow.java:3489)
            at com.android.internal.policy.impl.PhoneWindow.installDecor(PhoneWindow.java:3561)
            at com.android.internal.policy.impl.PhoneWindow.setContentView(PhoneWindow.java:367)
            at android.app.Activity.setContentView(Activity.java:2144)
            at myapp.com.app.Signup.onCreate(Signup.java:63)
            at android.app.Activity.performCreate(Activity.java:5933)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
            at android.app.ActivityThread.access$800(ActivityThread.java:144)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
     Caused by: java.lang.reflect.InvocationTargetException
            at java.lang.reflect.Constructor.newInstance(Native Method)
            at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
            at android.view.LayoutInflater.createView(LayoutInflater.java:607)
            at cat.ppicas.customtypeface.CustomTypeface.createView(CustomTypeface.java:285)
            at cat.ppicas.customtypeface.CustomTypeface.onCreateView(CustomTypeface.java:307)
            at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:806)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:504)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:414)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:365)
            at com.android.internal.policy.impl.PhoneWindow.generateLayout(PhoneWindow.java:3489)
            at com.android.internal.policy.impl.PhoneWindow.installDecor(PhoneWindow.java:3561)
            at com.android.internal.policy.impl.PhoneWindow.setContentView(PhoneWindow.java:367)
            at android.app.Activity.setContentView(Activity.java:2144)
            at myapp.com.app.Signup.onCreate(Signup.java:63)
            at android.app.Activity.performCreate(Activity.java:5933)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
            at android.app.ActivityThread.access$800(ActivityThread.java:144)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.TypedArray android.content.Context.obtainStyledAttributes(android.util.AttributeSet, int[], int, int)' on a null object reference
            at android.view.ViewStub.<init>(ViewStub.java:105)
            at android.view.ViewStub.<init>(ViewStub.java:101)
            at android.view.ViewStub.<init>(ViewStub.java:96)
            at java.lang.reflect.Constructor.newInstance(Native Method)
            at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
            at android.view.LayoutInflater.createView(LayoutInflater.java:607)
            at cat.ppicas.customtypeface.CustomTypeface.createView(CustomTypeface.java:285)
            at cat.ppicas.customtypeface.CustomTypeface.onCreateView(CustomTypeface.java:307)
            at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:806)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:504)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:414)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:365)
            at com.android.internal.policy.impl.PhoneWindow.generateLayout(PhoneWindow.java:3489)
            at com.android.internal.policy.impl.PhoneWindow.installDecor(PhoneWindow.java:3561)
            at com.android.internal.policy.impl.PhoneWindow.setContentView(PhoneWindow.java:367)
            at android.app.Activity.setContentView(Activity.java:2144)
            at myapp.com.app.Signup.onCreate(Signup.java:63)
            at android.app.Activity.performCreate(Activity.java:5933)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
            at android.app.ActivityThread.access$800(ActivityThread.java:144)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
```

**Comment by jmodrako** [comment]
You co it like me:

@Override public View onCreateView(String name, Context context, AttributeSet attrs) { return ViewStub.class.getSimpleName().contains(name) ? super.onCreateView(name, context, attrs) : CustomTypeface.getInstance().createView(name, context, attrs); }

Just wrap createView for all views except ViewStub. Works great!

**Comment by papiguy** [comment]
I have a similar issue for TextView as well. So clearly viewstub is not the only piece.

**Comment by Advice-Dog** [comment]
I think jmodrako's solution solves a problem if you're calling 

```
getLayoutInflater().setFactory(CustomTypeface.getInstance());
```

in an activity after calling it in another activity without calling finish();

I couldn't really find an solution, so I simply swapped over to Calligraphy, which uses the same structure ( allows you to apply styles to the stock TextViews, etc ) and doesn't seem to have any problems.

https://github.com/chrisjenx/Calligraphy

**Comment by ppicas** [comment]
Hi guys,

Thanks for sharing all this info. Finally I've been able to reproduce the issue. The new version 2.0.0 should solve the issue definitely.

The version changes from 1.0.1 to 2.0.0 because this version introduces changes that breaks the compatibility. I changed the current API because I figured out a better mechanism that solves the ViewStub issue and also ensures the compatibility with support libraries like appcompat-v7 and support-v4.
